### DISCUSSION TITLE
Option to expose more error detail in CI

### DISCUSSION THREAD

**Discussion by ArmaanMcleod** [microsoft/ps-rule#141] [Q&A] [answered]
Is there an option to expose more detail in CI(Github action in this case) when an error occurs?

For example, If I load two duplicate rules with different formats(JSON & YAML), I get this error in CI

![Screenshot 2022-02-06 215200](https://user-images.githubusercontent.com/20082136/152677454-2716ab38-35c8-42b0-bf17-a2cb0090d357.jpg)

I know this is because two rules are loaded with duplicate IDs, and an exception is thrown.

However, I would like to expose the full exception output in CI. Is this possible?

A user new to this tool will probably get stuck looking at an error like this, and we should provide an option to show the full error output in CI.

Apologise in advance if we do have this capability and I'm just not seeing it 😄.

**Comment by ElinoraWhite** [comment, marked as answer]
@ArmaanMcleod It probably relates to https://github.com/microsoft/ps-rule/blob/158611dd981a86aa5cb330b4c539e5889439c1d0/powershell.ps1#L257-L260.

Id' suggest we should be pushing a custom exception type through then we can provide some additional information in output.

**Reply by ArmaanMcleod** [reply]
@BernieWhite great idea, I was thinking the same this morning. I can raise an issue here for the custom exception type, and an issue in ps-rule to detect it and provide more error output.

**Comment by aavdberg** [comment]
I am running against the same problem that there is no detailed error message what is going precisely wrong.
